install-application/action.yml

```diff
@@ -0,0 +1,48 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+name: 'install-application'
+description: 'Power Platform Install Application'
+inputs:
+  environment-url:
+    description: 'URL of Power Platform environment; e.g. "https://test-env.crm.dynamics.com".'
+    required: true
+
+  environment-id:
+    description: 'Target environment application will be installed to.'
+    required: true
+
+  user-name:
+    description: 'Power Platform user name to authenticate with, e.g. myname@my-org.onmicrosoft.com. Setting this input makes user-name and password required; specifying alternate "app-id" credential set of inputs will result in an error.'
+    required: false
+
+  password-secret:
+    description: 'Power Platform password, required if authenticating with username. Do NOT checkin password, instead create a secret and reference it here with: see: https://help.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets#using-encrypted-secrets-in-a-workflow'
+    required: false
+
+  app-id:
+    description: 'The application id to authenticate with. Setting this input makes app-id, tenant-id and client-secret required; specifying alternate "username" credential set of inputs will result in an error.'
+    required: false
+
+  client-secret:
+    description: 'The client secret to authenticate with. Required if authenticating with app-id.'
+    required: false
+
+  tenant-id:
+    description: 'Tenant id if using app-id & client secret to authenticate.'
+    required: false
+
+  working-directory:
+    description: 'Working directory; default: root of repository.'
+    required: false
+
+  application-list:
```

Comment:
here's a dilemma: on the pac CLI side, the user can specify either a single app name OR a .json file with 1 or more app names.
-> for now: we only expose the application-list action input property, and make it required